mods/default/tools.lua

```diff
@@ -330,3 +330,138 @@ minetest.register_tool("default:sword_diamond", {
 		damage_groups = {fleshy=8},
 	}
 })
+
+minetest.register_tool("default:skeleton_key", {
+	description = "Skeleton Key",
+	inventory_image = "default_key_skeleton.png",
+	groups = { key=1 },
+	on_place = function(itemstack, placer, pointed_thing)
+		if pointed_thing.type ~= "node" then
+			return nil
+		end
+
+		local pos = pointed_thing.under
+		local node = minetest.get_node(pos)
+
+		if not node or node.name == "ignore" then
+			return nil
+		end
+
+		local type = ""
+		local owner = ""
+		local target_meta = nil
+		-- must be a lockable item
+		if node.name == "default:chest_locked" then
```

Comment:
Couldn't this be a field in the node definition instead? It would also allow mods to easily register new kinds of locked nodes, that can be shared too.